Multisig/Data/Services/Notification Service/RegisterNotificationTokenRequest.swift

```diff
@@ -50,11 +50,14 @@ struct RegisterNotificationTokenRequest: JSONRequest {
             self.version,
             self.deviceType,
             self.buildNumber,
-            self.timestamp
+            self.timestamp ?? ""
         ]
         .joined()
 
         if let signature = try? Signer.sign(string).value {
+            guard timestamp != nil else {
+                throw "'timestamp' parameter is required if signing key exists"
```

Comment:
maybe `preconditionFailure()` is better because this would be a programmer's error

Reply:
ok